backend/dal/Helpers/Extensions/BuildingExtensions.cs

```diff
@@ -41,7 +41,7 @@ public static IQueryable<Entity.Building> GenerateQuery(this PimsContext context
             filter.ThrowIfNull(nameof(user));
 
             // Check if user has the ability to view sensitive properties.
-            var userAgencies = user.GetAgencies();
+            var userAgencies = user.GetAgenciesAsNullable();
```

Comment:
Properties will not belong to an agency when they are disposed externally.  The `AgencyId` needed to be changed to nullable.